src/phoenix/server/api/queries.py

```diff
@@ -484,37 +489,38 @@ async def compare_experiments(
             base_experiment = next(
                 experiment for experiment in experiments if experiment.id == base_experiment_rowid
             )
-            revision_ids = (
-                select(func.max(models.DatasetExampleRevision.id))
-                .join(
-                    models.DatasetExample,
-                    models.DatasetExample.id == models.DatasetExampleRevision.dataset_example_id,
-                )
-                .where(
-                    and_(
-                        models.DatasetExampleRevision.dataset_version_id
-                        <= base_experiment.dataset_version_id,
-                        models.DatasetExample.dataset_id == base_experiment.dataset_id,
-                    )
+
+            base_experiment_split_ids_subquery = select(
+                models.ExperimentDatasetSplit.dataset_split_id
+            ).where(models.ExperimentDatasetSplit.experiment_id == base_experiment_rowid)
+
+            base_experiment_split_ids = (
+                await session.scalars(base_experiment_split_ids_subquery)
+            ).all()
+
+            # Use base experiment's split IDs for filtering (None if base has no splits)
+            split_ids_to_use = (
+                list(base_experiment_split_ids) if base_experiment_split_ids else None
+            )
+            # Get the revision IDs using the helper
+            revision_ids_query = (
+                get_dataset_example_revisions(
+                    base_experiment.dataset_version_id,
+                    dataset_id=base_experiment.dataset_id,
+                    split_ids=split_ids_to_use,
                 )
-                .group_by(models.DatasetExampleRevision.dataset_example_id)
+                .with_only_columns(models.DatasetExampleRevision.dataset_example_id)
                 .scalar_subquery()
             )
+
+            # Now build the examples query using those revision IDs
             examples_query = (
                 select(models.DatasetExample)
-                .distinct(models.DatasetExample.id)
-                .join(
-                    models.DatasetExampleRevision,
-                    onclause=and_(
-                        models.DatasetExample.id
-                        == models.DatasetExampleRevision.dataset_example_id,
-                        models.DatasetExampleRevision.id.in_(revision_ids),
-                        models.DatasetExampleRevision.revision_kind != "DELETE",
-                    ),
-                )
+                .where(models.DatasetExample.id.in_(revision_ids_query))
```

Comment:
why not do it this way?
```suggestion
                .join(models.ExperimentDatasetExample)
                .where(models.ExperimentDatasetExample.experiment_id == base_experiment_rowid)
```